sdk/keyvault/keyvault-common/src/keyVaultAuthenticationPolicy.ts

```diff
@@ -138,7 +152,9 @@ export function createKeyVaultChallengeCallbacks(
 
     const challenge = response.headers.get("WWW-Authenticate");
     if (!challenge) {
-      throw new Error("Missing challenge.");
+      throw new Error(
+        "A 401 response from the service must have a WWW-Authenticate header. This is a service bug.",
```

Comment:
Could another policy have gobbled the header? I'd probably just return the response in this case but if you want to throw, I suggest not assigning ownership of the bug

Reply:
Yeah maybe we can just return. I added a log message though just in case we get into this (very unlikely) place.